### PR TITLE
Android notifcations for Tab

### PR DESCRIPTION
**Pull request by redfast00**
Android notifications for Tab :tada:

- [X] Add AndroidDeviceRegistrationToken model belonging to user
- [x] Add an API for adding AndroidDeviceRegistrationTokens for a user
- [x] Add and configure [rpush](https://github.com/rpush/rpush) (a Ruby push notification library)
- [x] Add some business logic for when to send a notification

### CODE REVIEW

File: app/models/notification.rb [hunk 1]
```diff
@@ -12,10 +12,38 @@
...
+        app.name = "tappb"
+        app.auth_key = Rails.application.secrets.fcm_secret
+        app.connections = 1
+        app.save!
+      end
+
+      n = Rpush::Gcm::Notification.new
+      n.app = Rpush::Gcm::App.find_by_name("tappb")
+      n.registration_ids = user.android_device_registration_tokens.all.map{|r| r.token}
+      n.data = { message: message }
```
Comment (rien) [inline app/models/notification.rb]: For clarity, you could use `@message` for instance variables.

File: config/secrets.yml [hunk 2]
```diff
@@ -24,3 +24,4 @@ production:
   secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
   omniauth_client_id: ""
   omniauth_client_secret: ""
+  fcm_secret: ""
```
Comment (rien) [inline config/secrets.yml]: Shouldn't this be `gcm_secret`? (Google cloud messaging)

File: app/models/notification.rb [hunk 3]
```diff
@@ -12,10 +12,38 @@
...
+        app = Rpush::Gcm::App.new
+        app.name = "tappb"
+        app.auth_key = Rails.application.secrets.fcm_secret
+        app.connections = 1
+        app.save!
+      end
+
+      n = Rpush::Gcm::Notification.new
+      n.app = Rpush::Gcm::App.find_by_name("tappb")
+      n.registration_ids = user.android_device_registration_tokens.all.map{|r| r.token}
```
[the remark this answers is not in the document]
Comment (redfast00) [reply app/models/notification.rb]: The android_device_registration_tokens come from the app: when the app is started for the first time, it generates a token. We then send it to Tab via the `add_registration_token` API

File: app/models/notification.rb (hunk 1 above)
Comment (redfast00) [reply app/models/notification.rb]: I've tried that, but it didn't seem to work for some reason (message was nil)

File: config/secrets.yml (hunk 2 above)
Comment (redfast00) [reply config/secrets.yml]: Well, maybe, we're using Firebase Cloud Messaging, but the API is the same

File: app/models/notification.rb (hunk 1 above)
Comment (rien) [reply app/models/notification.rb]: Ah yes, because `message` is a method . `self.message` would be better then.

**Pull request merged**